### PR TITLE
Add support for probing podman auth files.

### PR DESCRIPTION
**Pull request by tmds**
@mthalman @baronfel ptal.

For reference: https://man.archlinux.org/man/containers-auth.json.5

### CODE REVIEW

**Comment by tmds** [comment]
@mthalman CI was not happy with my change to use C# 11. I've changed it back to C# 10. It should pass now.

**Comment by tmds** [comment]
@mthalman can you take a look?

This is part making the .NET SDK container tooling work well with `podman`.
Without this, the auth file created by `podman login` is not found.

File: src/Valleysoft.DockerCredsProvider/CredsProvider.cs [hunk 1]
```diff
@@ -22,65 +22,95 @@ internal static async Task<DockerCredentials> GetCredentialsAsync(string registr
...
-            } else {
-                return Path.Combine(
-                    env.GetFolderPath(Environment.SpecialFolder.UserProfile),
-                    ".docker",
-                    "config.json");
-            }
-    }
-
-    private static async Task<ICredStore> GetCredStoreAsync(string registry, IFileSystem fileSystem, IProcessService processService, IEnvironment environment)
+    internal static string[] GetConfigFilePaths(IEnvironment env)
```
Comment (mthalman) [inline src/Valleysoft.DockerCredsProvider/CredsProvider.cs:25]:
```suggestion
    /// <summary>
    /// Returns a set of candidate file paths for the config file that should be checked in the order listed (i.e. they are listed in priority order).
    /// </summary>
    internal static string[] GetConfigFilePaths(IEnvironment env)
```

**Review by mthalman — APPROVED** [review]
Just one suggestion for code comments.

**Comment by tmds** [comment]
Thanks for the review!

Once this is merged, can you make a new release, so it can be used by the dotnet/sdk repo?

**Pull request merged**